test/e2e/src/util.js

```diff
@@ -1,64 +1,89 @@
 module.exports.rpcUrl = rpcUrl
 const networkPassphrase = process.env.SOROBAN_NETWORK_PASSPHRASE ?? "Standalone Network ; February 2017";
 module.exports.networkPassphrase = networkPassphrase
+const friendbotUrl = process.env.SOROBAN_FRIENDBOT_URL ?? "http://localhost:8000/friendbot";
+module.exports.friendbotUrl = friendbotUrl
 
-class Wallet {
-  constructor(publicKey) {
-    this.publicKey = publicKey
-  }
+async function generateFundedKeypair() {
+  const keypair = Keypair.random()
+  await fetch(`${friendbotUrl}/friendbot?addr=${keypair.publicKey()}`)
+  return keypair
+};
+module.exports.generateFundedKeypair = generateFundedKeypair
 
-  isConnected = () => Promise.resolve(true) // eslint-disable-line class-methods-use-this
+/**
+ * Generates a ContractClient for the contract with the given name.
+ * Also generates a new account to use as as the keypair of this contract. This
+ * account is funded by friendbot. You can pass in an account to re-use the
+ * same account with multiple contract clients.
+ *
+ * By default, will re-deploy the contract every time. Pass in the same
+ * `contractId` again if you want to re-use the a contract instance.
+ */
+async function clientFor(contract, { keypair = generateFundedKeypair(), contractId } = {}) {
+  if (!contracts[contract]) {
+    throw new Error(
+      `Contract ${contract} not found. ` +
+      `Pick one of: ${Object.keys(contracts).join(", ")}`
+    )
+  }
 
-  isAllowed = () => Promise.resolve(true) // eslint-disable-line class-methods-use-this
+  keypair = await keypair // eslint-disable-line no-param-reassign
+  const wallet = new SorobanRpc.ExampleNodeWallet(keypair, networkPassphrase)
 
-  getUserInfo = () => Promise.resolve({ publicKey: this.publicKey })
+  const spec = new ContractSpec(contracts[contract].xdr)
 
-  signTransaction = async (tx) => {
-    const t = TransactionBuilder.fromXDR(tx, networkPassphrase);
-    t.sign(getKeypair(this.publicKey));
-    return t.toXDR();
+  const wasmHash = contracts[contract].hash;
+  if (!wasmHash) {
+    throw new Error(`No wasm hash found for \`contracts[${contract}]\`! ${JSON.stringify(contracts[contract], null, 2)}`)
   }
 
-  signAuthEntry = async (entryXdr, opts) => (
-    getKeypair(opts?.accountToSign ?? this.publicKey)
-      .sign(hash(Buffer.from(entryXdr, "base64")))
-      .toString('base64')
-  )
-}
-module.exports.Wallet = Wallet
+  // TODO: do this with js-stellar-sdk, instead of shelling out to the CLI
+  contractId = contractId ?? spawnSync("./target/bin/soroban", [ // eslint-disable-line no-param-reassign
+    "contract",
+    "deploy",
+    "--source",
+    keypair.secret(),
+    "--wasm-hash",
+    wasmHash,
+  ], { shell: true, encoding: "utf8" }).stdout.trim();
 
-module.exports.wallet = new Wallet(root.keypair.publicKey())
+  const client = spec.generateContractClient({
+    networkPassphrase,
+    contractId,
+    rpcUrl,
+    allowHttp: true,
+    publicKey: keypair.publicKey(),
+    ...wallet,
```

Comment:
Note that the exposed interface no longer uses the word `wallet` at all, but people may still choose to name the thing which exports `signTransaction` and `signAuthEntry` a _wallet_.